### PR TITLE
 fix: support `.codecov.yml`/`codecov.yml` files for token

### PR DESCRIPTION
**Pull request by hellatan**
currently, `<rootDir>/.codecov.yml` and `<rootDir>/.codecov.yml` are not recognized by this node package. it looks like `query.yaml` was getting blown away when `query = detectProvider()` was called since currently `query.yaml` is assigned above it. 

I moved the `query.yaml` assignment after the `query = detectProvider()` call to maintain the yaml file reference.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,8 @@
   "dependencies": {
     "argv": "^0.0.2",
     "ignore-walk": "^3.0.1",
+    "js-yaml": "^3.12.0",
+    "mock-fs": "^4.6.0",
```
Comment (eddiemoore) [inline package.json]: This should be in the dev dependencies.

File: package.json [hunk 2]
```diff
@@ -36,8 +36,10 @@
     "eslint-config-prettier": "^2.9.0",
     "expect.js": "^0.3.1",
     "husky": "^0.14.3",
+    "js-yaml": "^3.12.0",
```
Comment (eddiemoore) [inline package.json]: Sorry I should have been more specific before. Only the `mock-fs` needs to be in devDeps. `js-yaml` would need to be in normal `dependencies` as it's being used in the main library file.

Comment (hellatan) [reply package.json]: oh duh...sorry wasn't thinking. updated.

**Comment by hellatan** [comment]
ugh...seems like some sort of timeout issue with the tests:
```
// node 10
  1) Codecov
       upload v2 doesn't throw runtime error:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/travis/build/codecov/codecov-node/test/upload.js)
```
and
```
// node 5
  1) Codecov
       can remove files after uploading:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/travis/build/codecov/codecov-node/test/upload.js)
```
i'll try rerunning this build.

**Comment by hellatan** [comment]
@eddiemoore ok seems like it was just a network issue WRT those failing tests. Can this be merged soon?

**Comment by eddiemoore** [comment]
Released in 3.1.0

**Pull request merged**